x/posts/internal/types/media_post.go

```diff
@@ -0,0 +1,260 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type MediaPost struct {
+	TextPost
+	Medias PostMedias `json:"medias"`
+}
+
+func NewMediaPost(post TextPost, medias PostMedias) MediaPost {
+	return MediaPost{
+		TextPost: post,
+		Medias:   medias,
+	}
+}
+
+// String implements fmt.Stringer
+func (mp MediaPost) String() string {
+	bytes, err := json.Marshal(&mp)
+	if err != nil {
+		panic(err)
+	}
+
+	return string(bytes)
+}
+
+// GetID implements Post GetID
+func (mp MediaPost) GetID() PostID {
+	return mp.PostID
+}
+
+// GetParentID implements Post GetParentID
+func (mp MediaPost) GetParentID() PostID {
+	return mp.ParentID
+}
+
+func (mp MediaPost) SetMessage(message string) Post {
+	mp.Message = message
+	return mp
+}
+
+func (mp MediaPost) GetMessage() string {
+	return mp.Message
+}
+
+func (mp MediaPost) CreationTime() time.Time {
+	return mp.Created
+}
+
+func (mp MediaPost) SetEditTime(time time.Time) Post {
+	mp.LastEdited = time
+	return mp
+}
+
+func (mp MediaPost) GetEditTime() time.Time {
+	return mp.LastEdited
+}
+
+func (mp MediaPost) CanComment() bool {
+	return mp.AllowsComments
+}
+
+func (mp MediaPost) GetSubspace() string {
+	return mp.Subspace
+}
+
+func (mp MediaPost) GetOptionalData() map[string]string {
+	return mp.OptionalData
+}
+
+func (mp MediaPost) Owner() sdk.AccAddress {
+	return mp.Creator
+}
+
+// Validate implements Post Validate
+func (mp MediaPost) Validate() error {
+	if err := mp.TextPost.Validate(); err != nil {
+		return err
+	}
+
+	for _, media := range mp.Medias {
+		if err := media.Validate(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (mp MediaPost) Equals(other Post) bool {
+	// Cast and delegate
+	if otherMp, ok := other.(MediaPost); ok {
+		return checkMediaPostEquals(mp, otherMp)
+	}
+
+	return false
+}
+
+// Equals implements Post Equals
+func checkMediaPostEquals(first MediaPost, second MediaPost) bool {
+	if !first.TextPost.Equals(second.TextPost) {
+		return false
+	}
+
+	if len(first.Medias) != len(second.Medias) {
+		return false
+	}
+
+	for index, media := range first.Medias {
+		if media != second.Medias[index] {
+			return false
+		}
+	}
```

Comment:
I would prefer we review this putting all the `Medias` equality checking inside the `Medias` type instead. 

This would allows to simplify this method to be

```golang
return first.TextPost.Equals(second.TextPost) && first.Medias.Equals(second.Medias)
```